plugins/aws-appmesh/plugin/commands.go

```diff
@@ -214,29 +215,10 @@ func (plugin *Plugin) setupIngressRules(
 	if config.ProxyIngressPort == "" || len(config.AppPorts) == 0 {
 		return nil
 	}
-
-	err := iptable.NewChain("nat", ingressChain)
-	if err != nil {
-		return err
+	var interceptPortToListenerPorts = map[string]string{
+		config.ProxyIngressPort: config.AppPorts,
 	}
-
-	// Route everything arriving at the application port to proxy.
-	err = iptable.Append("nat", ingressChain, "-p", "tcp", "-m", "multiport", "--dports", config.AppPorts,
-		"-j", "REDIRECT", "--to-port", config.ProxyIngressPort)
-	if err != nil {
-		log.Errorf("Append rule to redirect traffic to proxyIngressPort failed: %v", err)
-		return err
-	}
-
-	// Apply ingress chain to everything non-local.
-	err = iptable.Append("nat", "PREROUTING", "-p", "tcp", "-m", "addrtype", "!", "--src-type",
-		"LOCAL", "-j", ingressChain)
-	if err != nil {
-		log.Errorf("Append rule to jump from PREROUTING to ingress chain failed: %v", err)
-		return err
-	}
-
-	return nil
+	return pluginutils.RedirectNonLocalTraffic(iptable, ingressChain, interceptPortToListenerPorts)
```

Comment:
Do we think these utility/wrapper functions provide any benefits? I would understand if this was a complicated piece of code that we wanted to reuse across plugins, but what the function does is very straightforward. This decreases readability of plugins because the reader now has to look at three different packages when reading and making changes.

I'd much rather keep the aws-appmesh plugin as is.

Reply:
I agree that there is not a lot of common code but having these methods would help reuse them in a new plugin in the future rather than identifying which table we should be using for port redirection, which options we need to set, etc. But if you believe this is not needed now, I can revert back.

Reply:
Published a new revision handling all other comments except this one.